packages/discord.js/src/structures/SelectMenuBuilder.js

```diff
@@ -2,10 +2,19 @@
 
 const { SelectMenuBuilder: BuildersSelectMenuComponent, isJSONEncodable } = require('@discordjs/builders');
 const Transformers = require('../util/Transformers');
+const Util = require('../util/Util');
 
 class SelectMenuBuilder extends BuildersSelectMenuComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+  constructor({ options, ...data }) {
+    super(
+      Transformers.toSnakeCase({
+        options: options.map(({ emoji, ...option }) => ({
+          emoji: emoji && typeof emoji === 'string' ? Util.parseEmoji(emoji) : emoji,
+          ...option,
```

Comment:
These two lines should go the other way around (as spreading options will replace the emoji you just made)